From: Iain Lane <iain.lane@canonical.com>
Date: Thu, 6 Sep 2018 18:00:00 +0100
Subject: Revert "gdm-wayland-session,gdm-x-session: register after delay"

This commit breaks wayland -> Xorg fallback.

This reverts commit 708618746683ea086068e2498fd1c69917c6189a.

Bug-Upstream: https://gitlab.gnome.org/GNOME/gdm/issues/419
---
 daemon/gdm-wayland-session.c | 23 +++++++----------------
 daemon/gdm-x-session.c       | 25 ++++++++-----------------
 2 files changed, 15 insertions(+), 33 deletions(-)

diff --git a/daemon/gdm-wayland-session.c b/daemon/gdm-wayland-session.c
index de1991b..94f49e1 100644
--- a/daemon/gdm-wayland-session.c
+++ b/daemon/gdm-wayland-session.c
@@ -454,21 +454,6 @@ on_sigterm (State *state)
         return G_SOURCE_CONTINUE;
 }
 
-static gboolean
-on_registration_delay_complete (State *state)
-{
-        gboolean ret;
-
-        ret = register_display (state, state->cancellable);
-
-        if (!ret) {
-                g_printerr ("Unable to register display with display manager\n");
-                g_main_loop_quit (state->main_loop);
-        }
-
-        return G_SOURCE_REMOVE;
-}
-
 int
 main (int    argc,
       char **argv)
@@ -543,7 +528,13 @@ main (int    argc,
                 goto out;
         }
 
-        g_timeout_add_seconds (2, (GSourceFunc) on_registration_delay_complete, state);
+        ret = register_display (state, state->cancellable);
+
+        if (!ret) {
+                g_printerr ("Unable to register display with display manager\n");
+                exit_status = EX_SOFTWARE;
+                goto out;
+        }
 
         g_main_loop_run (state->main_loop);
 
diff --git a/daemon/gdm-x-session.c b/daemon/gdm-x-session.c
index 412999c..3b2fcef 100644
--- a/daemon/gdm-x-session.c
+++ b/daemon/gdm-x-session.c
@@ -810,21 +810,6 @@ on_sigterm (State *state)
         return G_SOURCE_CONTINUE;
 }
 
-static gboolean
-on_registration_delay_complete (State *state)
-{
-        gboolean ret;
-
-        ret = register_display (state, state->cancellable);
-
-        if (!ret) {
-                g_printerr ("Unable to register display with display manager\n");
-                g_main_loop_quit (state->main_loop);
-        }
-
-        return G_SOURCE_REMOVE;
-}
-
 int
 main (int    argc,
       char **argv)
@@ -911,6 +896,14 @@ main (int    argc,
                 goto out;
         }
 
+        ret = register_display (state, state->cancellable);
+
+        if (!ret) {
+                g_printerr ("Unable to register display with display manager\n");
+                exit_status = EX_SOFTWARE;
+                goto out;
+        }
+
         ret = spawn_session (state, run_script, state->cancellable);
 
         if (!ret) {
@@ -919,8 +912,6 @@ main (int    argc,
                 goto out;
         }
 
-        g_timeout_add_seconds (2, (GSourceFunc) on_registration_delay_complete, state);
-
         g_main_loop_run (state->main_loop);
 
         /* Only use exit status of session if we're here because it exit */
